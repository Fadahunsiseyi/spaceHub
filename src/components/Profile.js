import React from 'react';
import { useSelector } from 'react-redux';

const Profile = () => {
  const dragons = useSelector((state) => state.dragonReducer);
  const rockets = useSelector((state) => state.rocketReducers);
  const missions = useSelector((state) => state.missionReducer);
  const activeDragons = dragons.dragons.filter(
    (dragon) => dragon.reserve === true,
  );
  const activeRockets = rockets.rockets.filter(
    (rocket) => rocket.reserve === true,
  );
  const activeMissions = missions.filter((mission) => mission.join);
  return (
    <div className="wrapper_profile">
      <div className="rockets-profile">
        <h2>My Rockets</h2>
        <ul className="lists">
          {activeRockets.length > 0 ? (
            activeRockets.map((rock) => (
              <li key={rock.rocket_name}>{rock.rocket_name}</li>
            ))
          ) : (
            <li>
              <p>Reserved Rockets Not Found</p>
            </li>
          )}
        </ul>
      </div>
      <div className="rockets-profile">
        <h2>My Dragons</h2>
        <ul className="lists">
          {activeDragons.length > 0 ? (
            activeDragons.map((drag) => (
              <li key={drag.name}>{drag.name}</li>
            ))
          ) : (
            <li>
              <p>Reserved Rockets Not Found</p>
            </li>
          )}
        </ul>
      </div>
      <div className="mission-profile">
        <h2>My Missions</h2>
        <ul className="lists">
          {activeMissions.length > 0 ? (
            activeMissions.map((mission) => (
              <li key={mission.mission_name}>{mission.mission_name}</li>
            ))
          ) : (
            <li>
              <p>Activated Missions Not Found</p>
            </li>
          )}
        </ul>
      </div>
    </div>
  );
};

export default Profile;
