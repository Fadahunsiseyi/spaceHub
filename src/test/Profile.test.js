import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Profile from '../components/Profile';

describe('To test Profile components', () => {
  it('To Test Profile components renders successfully!', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
