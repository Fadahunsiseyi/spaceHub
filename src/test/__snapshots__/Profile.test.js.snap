// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To test Profile components To Test Profile components renders successfully! 1`] = `
<div
  className="wrapper_profile"
>
  <div
    className="rockets-profile"
  >
    <h2>
      My Rockets
    </h2>
    <ul
      className="lists"
    >
      <li>
        <p>
          Reserved Rockets Not Found
        </p>
      </li>
    </ul>
  </div>
  <div
    className="mission-profile"
  >
    <h2>
      My Missions
    </h2>
    <ul
      className="lists"
    >
      <li>
        <p>
          Activated Missions Not Found
        </p>
      </li>
    </ul>
  </div>
</div>
`;
